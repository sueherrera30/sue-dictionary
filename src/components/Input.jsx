import { useState } from 'react';
import { useNavigate } from "react-router-dom";
import styles from './styles/input.module.css';
import book from '../assets/book.png';
const Input = () => {
    const [word, setWord] = useState('');
    const[message, setMessage] = useState({
        state: false,
        content: '',
    })
    let navigate = useNavigate();

    const saveValue = (event) => {
        event.preventDefault();
        const { value } = event.target;
        setWord(value);
    };
    const handleSearch = (event) => {
        if (event.key === 'Enter')  {
            if(word === '') setMessage(() => {
                return {
                state: true,
                content: 'please add a word!'
            }})
            else if(word.split(' ').length > 1) setMessage(() => { 
                return {
                state: true,
                content: 'please add just 1 word'
            }})
            else {
                navigate(`/definition/${word}`)  
            }
        }
        if(event.key === 'Backspace') {
            setMessage(() => {
                return {
                state: false,
                content: ''
            }})
        }
    };

    return (
        <div className={styles.mainContainer}>
            <div className={styles.imageContainer}>
                <img src={book} alt="main" className={styles.image}/>
            </div>
            <div>
                <span className={styles.input}>
                    <input
                        value={word}
                        onChange={saveValue}
                        onKeyDown={handleSearch}
                        type="text"
                        placeholder="write a word..."
                    />
                    <span></span>
                </span>
                {message.state && <p className={styles.message}>{message.content}</p>}
            </div>
        </div>
    )
}

export default Input;