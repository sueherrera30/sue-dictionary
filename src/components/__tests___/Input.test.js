import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import {  MemoryRouter } from 'react-router-dom';

import Input from '../Input';

afterEach(() => {
    cleanup();
});

const mockedUsedNavigate = jest.fn();
const handleSearchMocked = jest.fn(() => mockedUsedNavigate);
const saveValueMocked = jest.fn();
const wordMocked = 'hello';
const message = {
    state: false,
    content: 'please add a word!',
}

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
  }));

  describe('testing input component', () => {  
    let inputComponent = <MemoryRouter>
            <Input handleSearch={handleSearchMocked} saveValue={saveValueMocked} word={wordMocked} message={message} />
        </MemoryRouter>;
        test('<Input />', () => {
            render(inputComponent);
            expect(screen.container).toMatchSnapshot();
            expect(screen.getByTestId('input').tagName).toBe('INPUT');
        });

        test('Goes to definition page when enter', async () => {
            render(inputComponent);
            const input = screen.getByTestId("input");
            fireEvent.change(input, { target: { value: 'hello' } });
            fireEvent.keyPress(input, { key: "Enter", code: 13, charCode: 13});
       
            // const inputvalue = screen.getByTestId('input').value;
            // await fireEvent.keyDown(screen.getByTestId('input'), {key: 'enter', keyCode: 13})
            // expect(mockedUsedNavigate).toHaveBeenCalledWith(`/definition/${inputvalue}`)
            // expect(handleSearchMocked.mock.calls.length).toBe(1);
            // expect(mockedUsedNavigate).toHaveBeenCalledTimes(1)
        });
        // test('save word in enter', async () => {
        //     render(inputComponent);
        //     await fireEvent.change(screen.getByTestId('input'), { target: { value: 'hello'}})
        //     expect(enterMockfuntion).toHaveBeenCalledTimes(1);
        //     expect(enterMockfuntion).toHaveBeenCalledWith({
        //         content: 'hello',
        //     });
        // });

  });