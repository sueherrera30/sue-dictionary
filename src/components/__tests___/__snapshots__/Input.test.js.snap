// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing input component <Input /> 1`] = `undefined`;
